### PR TITLE
3394 - Send email to inviting user

### PR DESCRIPTION
**Pull request by yaguzmang**
- Adding migration for the new `invited_by_user_uuid` column in `users_role`, and adding the logic to notify the inviting user when the invitation is accepted. 

- Adding ar, en, fr, ru, zh translations, and added my Spanish translation as a draft while we get the official one (to be modified in subsequent PR if needed).

### CODE REVIEW

File: src/i18n/resources/fr.js [hunk 1]
```diff
@@ -480,6 +480,22 @@ L'équipe de FRA
...
+
+L’équipe FRA
+    `,
+      htmlMessage: `Cher {{recipientName}} {{recipientSurname}},
+<br/><br/>
+L’utilisateur {{invitedUserName}} {{invitedUserSurname}} a accepté votre invitation à participer à {{assessmentName}} {{cycleName}} en tant que {{role}} pour le/la <b><a href="{{- manageCollaboratorsUrl}}">{{country}}</a></b>.
+<br/><br/>
+L’équipe FRA
+    `,
+    },
```
Comment (yaguzmang) [inline src/i18n/resources/fr.js]: In the received email for the translation, we have:
"**Cher [for a man]/Chère [for a women]** {{recipient name}} {{recipient surname}}, ..."
But based on the other translations we have, like in the [invitation email](https://github.com/openforis/fra-platform/pull/3515/commits/5faeef68a4f8356986c3f096fb7cf6d78328d3cc#diff-73afee1ef09cb9f4f1a71b6be625a7ea1726173308b8d1e2f0383daaaa766182L497), we are only using "Cher". Should I add any logic to handle this using, maybe, the appellation?

**Comment by sorja** [comment]
when altering public tables you must alter them both in migration-steps and migrations
here is an example:

creating a migration for `public.file`:
https://github.com/openforis/fra-platform/pull/3404/files#diff-40958ad7ea2e1252c6768653933e674b6491b379d8d328e4bba4d762bd0fd2cc

update src/server/repository/assessment/assessment/getCreateSchemaDDL.ts if needed:
https://github.com/openforis/fra-platform/pull/3404/files#diff-7ea7a9e0804acaec74c94ad59932ddb1e16c3bfff0d6cca1eb9f396b011f9dd7

add migration-step to alter existing db:
https://github.com/openforis/fra-platform/pull/3404/files#diff-da3f29e665ada58f5eafdc2bd8c6e06473fbebc2fc07150f8ad03bd2bbc3ac95

hope this helps 🥇

**Comment by sorja** [comment]
looks good to me

File: src/i18n/resources/fr.js [hunk 2]
```diff
@@ -480,6 +480,22 @@ L'équipe de FRA
     invitationLink: "Lien d'invitation",
     sendInvitation: "Envoyer un email d'invitation",
     inviteAgain: 'Inviter de nouveau',
+    invitationAcceptedNotification: {
```
Comment (sorja) [inline src/i18n/resources/fr.js]: i would introduce new file and/or folder (you can see what is better) for emails:
en/emails/<email>
or
en/email.js

used as

`email.userInvitation.acceptedNotification: { ... }`

File: src/i18n/resources/fr.js (hunk 1 above)
Comment (minotogna) [reply src/i18n/resources/fr.js]: naaa just use Cher/Chère.

File: src/i18n/resources/fr.js (hunk 2 above)
Comment (minotogna) [reply src/i18n/resources/fr.js]: sounds good actually . we'll refactor later on other e-mail messages.

File: src/server/controller/user/acceptInvitation.ts [hunk 3]
```diff
@@ -42,6 +45,18 @@ export const acceptInvitation = async (
       t
     )
 
+    if (!Objects.isEmpty(userRole.invitedByUserUuid)) {
```
Comment (minotogna) [inline src/server/controller/user/acceptInvitation.ts:48]: I'm wondering, can this be ever null?

File: src/test/migrations/steps/20240213154855-step-add-invited_by_user_uuid-column-to-users_role.ts [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import { BaseProtocol } from 'server/db'
+
+export default async (client: BaseProtocol) => {
```
Comment (minotogna) [inline src/test/migrations/steps/20240213154855-step-add-invited_by_user_uuid-column-to-users_role.ts]: wait, if we run first migration and then migration-steps, does this work ?
I think only migration is necessary .

But here, you need to populate the invited_by_user_uuid.

File: src/server/controller/user/acceptInvitation.ts (hunk 3 above)
Comment (sorja) [reply src/server/controller/user/acceptInvitation.ts:48]: maybe on admin manually adding roles and not inviting?

File: src/i18n/resources/fr.js (hunk 2 above)
Comment (yaguzmang) [reply src/i18n/resources/fr.js]: en/email.js
then in email.js:
```
userInvitation: {
   invitation: {
     subject...
     textMessage...
   }
   invitationAccepted: {
     subject...
     textMessage...
   }
}
```

Comment (minotogna) [reply src/i18n/resources/fr.js]: let's just have inside email:

invitation :{},
invitationAccepted :{},

so each email a root entry in the file

**Review by minotogna — APPROVED** [review]

**Pull request merged**